### PR TITLE
Apid 405

### PR DESCRIPTION
**Pull request by srinivasalumunagala**

### CODE REVIEW

File: app/uk/gov/hmrc/pushpullnotificationsapi/controllers/BoxController.scala [hunk 1]
```diff
@@ -84,6 +84,20 @@ class BoxController @Inject()(validateUserAgentHeaderAction: ValidateUserAgentHe
         } (actualBody, manifest, RequestFormatters.createClientManagedBoxRequestFormatter) recover recovery
       }
 
+  def deleteClientManagedBox(boxId: BoxId): Action[AnyContent] =
+    (Action
+      andThen validateUserAgentHeaderAction
```
Comment (ghchapman) [inline app/uk/gov/hmrc/pushpullnotificationsapi/controllers/BoxController.scala:89]: I don't think we want this action on public APIs.

File: app/uk/gov/hmrc/pushpullnotificationsapi/repository/BoxRepository.scala [hunk 2]
```diff
@@ -94,11 +94,19 @@ class BoxRepository @Inject()(mongo: MongoComponent)
       case NonFatal(e) => Future.successful(BoxCreateFailedResult(e.getMessage))
     }
 
+  def deleteBox(passedBoxId: BoxId)(implicit ec: ExecutionContext): Future[DeleteBoxResult] =
```
Comment (ghchapman) [inline app/uk/gov/hmrc/pushpullnotificationsapi/repository/BoxRepository.scala:97]: I'd probably go with boxId, all params are passed 😄

File: it/uk/gov/hmrc/pushpullnotificationsapi/controllers/BoxControllerISpec.scala [hunk 3]
```diff
@@ -457,6 +466,41 @@ class BoxControllerISpec extends ServerBaseISpec
...
+
+    "failed to delete a CMB when not found and return status 404" in {
+      primeApplicationQueryEndpoint(Status.OK, tpaResponse, clientId.value)
+      primeAuthServiceSuccess(clientId.value, "{\"authorise\" : [ ], \"retrieve\" : [ \"clientId\" ]}")
+      await(repo.createBox(clientManagedBox))
+      val deleteResult = callDeleteClientManagedBoxEndpoint(clientManagedBox.boxId.raw.replaceAll("9", "8"), validHeadersWithAcceptHeader)
+      deleteResult.status shouldBe NOT_FOUND
+    }
+
+    "failed to delete a CMB when unauthorised and return status 404" in {
```
Comment (ghchapman) [inline it/uk/gov/hmrc/pushpullnotificationsapi/controllers/BoxControllerISpec.scala:494]: Test name is misleading as we're asserting on 403.

**Review by jameshall1999 — APPROVED** [review]

File: app/uk/gov/hmrc/pushpullnotificationsapi/services/BoxService.scala [hunk 4]
```diff
@@ -133,5 +140,15 @@ class BoxService @Inject()(repository: BoxRepository,
       })
   }
 
-
+  private def validateAndDeleteBox(box: Box, clientId: ClientId, boxId: BoxId)(implicit ec: ExecutionContext): Future[DeleteBoxResult] = {
+    if (box.boxCreator.clientId == clientId && box.clientManaged && box.boxId == boxId) {
```
Comment (ghchapman) [inline app/uk/gov/hmrc/pushpullnotificationsapi/services/BoxService.scala:144]: I think box.boxId == boxId is always true, as the boxId was used to retrieve the box in the first place.  We don't need the boxId param (as we can just get it from the box) and we don't need to do the box.boxId==boxId check,

**Comment by srinivasalumunagala** [comment]
dup

**Pull request closed without merging**